sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/Client/CryptographyClient.cs

```diff
@@ -0,0 +1,213 @@
+
+        /// <summary>
+        /// KeyVault Uri
+        /// </summary>
+        public Uri KeyVaultKeyUri { get; }
+
+        /// <summary>
+        /// KeyVault Key
+        /// </summary>
+        public Key KeyVaultKey { get; }
+        #endregion
+
+        #region Constructor
+        /// <summary>
+        /// Initializes a new instance of Cryptography Client
+        /// </summary>
+        protected CryptographyClient()
+        {
+            KeyVaultKeyUri = null;
+            KeyVaultKey = null;
+        }
+
+        #region Local Crypto
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="keyVaultKeyUri"></param>
+        public CryptographyClient(Uri keyVaultKeyUri)
+            : this(keyVaultKeyUri, new CryptographyClientOptions()) { }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="keyVaultKeyUri"></param>
+        /// <param name="options"></param>
+        public CryptographyClient(Uri keyVaultKeyUri, CryptographyClientOptions options)
+        {
+            KeyVaultKeyUri = keyVaultKeyUri;
+            options.EnableServerCryptographyOperations = false;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="keyVaultKey"></param>
+        public CryptographyClient(Key keyVaultKey)
+            : this(keyVaultKey, new CryptographyClientOptions()) { }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="keyVaultKey"></param>
+        /// <param name="options"></param>
+        public CryptographyClient(Key keyVaultKey, CryptographyClientOptions options)
+        {
+            KeyVaultKey = keyVaultKey;
+            options.EnableServerCryptographyOperations = false;
+        }
+
+        #endregion
+
+        #region Server Crypto
+
+        /// <summary>
+        /// Initializes a new instance of Cryptography Client
+        /// </summary>
+        /// <param name="keyVaultKeyUri">Endpoint URL for the Azure Key Vault service.</param>
+        /// <param name="credential">Represents a credential capable of providing an OAuth token.</param>
+        public CryptographyClient(Uri keyVaultKeyUri, TokenCredential credential)
+            : this(keyVaultKeyUri, credential, new CryptographyClientOptions()) { }
+
+        /// <summary>
+        /// Initializes a new instance of Cryptography Client
+        /// </summary>
+        /// <param name="keyVaultKeyUri">Endpoint URL for the Azure Key Vault service.</param>
+        /// <param name="credential">Represents a credential capable of providing an OAuth token.</param>
+        /// <param name="options">Options that allow to configure the management of the request sent to Key Vault.</param>
+        public CryptographyClient(Uri keyVaultKeyUri, TokenCredential credential, CryptographyClientOptions options) : this()
+        {
+            Check.NotNull(keyVaultKeyUri, nameof(keyVaultKeyUri));
```

Comment:
I'd rather we don't use helper methods for null checks because it adds another method to the stack so the user has to look up the stack to see what method null was passed to.  if you want to make it one line do something like

~~~ c#
keyId = keyId ?? throw new ArgumentNullException(nameof(keyId));
~~~

Reply:
So here is how the null check stack looks like
```
Result StackTrace:	
at Azure.Security.KeyVault.Cryptography.Utilities.Check.NotNull(Object argument, String argumentName) in D:\myFork\sdkForNet\mas\sdk\keyvault\Azure.Security.KeyVault.Keys\src\Cryptography\Utilities\Check.cs:line 52
   at Azure.Security.KeyVault.Cryptography.Client.CryptographyClient..ctor(Key keyVaultKey, TokenCredential credential, CryptographyClientOptions options) in D:\myFork\sdkForNet\mas\sdk\keyvault\Azure.Security.KeyVault.Keys\src\Cryptography\Client\CryptographyClient.cs:line 154
   at Azure.Security.KeyVault.Cryptography.Client.CryptographyClient..ctor(Key keyVaultKey, TokenCredential credential) in D:\myFork\sdkForNet\mas\sdk\keyvault\Azure.Security.KeyVault.Keys\src\Cryptography\Client\CryptographyClient.cs:line 144
   at Azure.Security.KeyVault.Cryptography.Tests.CryptoTestBase.InitCryptoClient(Key kvKey) in D:\myFork\sdkForNet\mas\sdk\keyvault\Azure.Security.KeyVault.Keys\tests\CryptoTests\KVCryptoTestBase.cs:line 104
   at Azure.Security.KeyVault.Cryptography.Tests.KVCryptoTests.InstantiateCryptoClient() in D:\myFork\sdkForNet\mas\sdk\keyvault\Azure.Security.KeyVault.Keys\tests\CryptoTests\KVCryptoTests.cs:line 36
   at NUnit.Framework.Internal.TaskAwaitAdapter.GenericAdapter`1.GetResult() in D:\a\1\s\src\NUnitFramework\framework\Internal\TaskAwaitAdapter.cs:line 99
   at NUnit.Framework.Internal.AsyncToSyncAdapter.Await(Func`1 invoke) in D:\a\1\s\src\NUnitFramework\framework\Internal\AsyncToSyncAdapter.cs:line 60
   at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute(TestExecutionContext context) in D:\a\1\s\src\NUnitFramework\framework\Internal\Commands\TestMethodCommand.cs:line 64
   at NUnit.Framework.Internal.Commands.BeforeAndAfterTestCommand.<>c__DisplayClass1_0.<Execute>b__0() in D:\a\1\s\src\NUnitFramework\framework\Internal\Commands\BeforeAndAfterTestCommand.cs:line 58
   at NUnit.Framework.Internal.Commands.BeforeAndAfterTestCommand.RunTestMethodInThreadAbortSafeZone(TestExecutionContext context, Action action) in D:\a\1\s\src\NUnitFramework\framework\Internal\Commands\BeforeAndAfterTestCommand.cs:line 73
Result Message:	
System.ArgumentNullException : Value cannot be null.
Parameter name: keyVaultKey provided is null
```
Which I think is quite intuitive, but one of the reason I wanted to use helpers is to standardize a way to throw standards exception across SDKs
e.g. standard way to throw NullRef, ArgumentException etc, so it doesn't have to this Utlity, but any utility that will check and throw a standard null ref message.

This way no one really have to write the same exact code again and again, plus when we change our standards to throw standard error strings, we update once and all SDKs absorb those changes.